src/Features/CSharp/Portable/CodeFixes/PopulateSwitch/PopulateSwitchCodeFixProvider.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Simplification;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.PopulateSwitch
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.AddLabelsToSwitch), Shared]
+    [ExtensionOrder(After = PredefinedCodeFixProviderNames.ImplementInterface)]
+    internal partial class PopulateSwitchCodeFixProvider : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds
+        {
+            get { return ImmutableArray.Create(IDEDiagnosticIds.PopulateSwitchDiagnosticId); }
+        }
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            return PopulateSwitchFixAllProvider.Instance;
+        }
+
+        private static SwitchStatementSyntax GetSwitchStatementNode(SyntaxNode root, TextSpan span)
+        {
+            var token = root.FindToken(span.Start);
+            if (!token.Span.IntersectsWith(span))
+            {
+                return null;
+            }
+
+            return (SwitchStatementSyntax)root.FindNode(span);
+        }
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var span = context.Span;
+            var cancellationToken = context.CancellationToken;
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var model = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var node = GetSwitchStatementNode(root, span);
+            if (node == null)
+            {
+                return;
+            }
+
+            context.RegisterCodeFix(
+                new MyCodeAction(
+                    CSharpFeaturesResources.AddSwitchLabels,
+                    (c) => AddMissingSwitchLabelsAsync(model, document, root, node)),
+                context.Diagnostics);
+        }
+
+        private static async Task<Document> AddMissingSwitchLabelsAsync(SemanticModel model, Document document, SyntaxNode root, SwitchStatementSyntax switchBlock)
+        {
+            var enumType = (INamedTypeSymbol)model.GetTypeInfo(switchBlock.Expression).Type;
```

Comment:
while typing code, one could possibly have switch without a valid expression as yet. do we need null checks here?